From c6b30ef2375d0f37b7cfc2474d43c2f1107b5b20 Mon Sep 17 00:00:00 2001
From: Andrew Boie <andrew.p.boie@intel.com>
Date: Fri, 12 Oct 2012 15:26:37 -0700
Subject: init: leave ro.serialno unset if not provided in command line

Some systems may use bootloaders which can't modify the kernel
command line to provide the unit serial number, and this must
be supplied later. In order to allow this we need to not set
ro.serialno to an empty value, otherwise it's impossible to set it
to the real serial number later in the boot process.

Change-Id: Ieeb085c19437286400729aa00d2463ac9f8cec84
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>
---
 init/init.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/init/init.c b/init/init.c
index 9c172ca..f325f61 100755
--- a/init/init.c
+++ b/init/init.c
@@ -656,7 +656,7 @@ static void export_kernel_boot_props(void)
         const char *dest_prop;
         const char *def_val;
     } prop_map[] = {
-        { "ro.boot.serialno", "ro.serialno", "", },
+        { "ro.boot.serialno", "ro.serialno", NULL, },
         { "ro.boot.mode", "ro.bootmode", "unknown", },
         { "ro.boot.baseband", "ro.baseband", "unknown", },
         { "ro.boot.bootloader", "ro.bootloader", "unknown", },
@@ -664,7 +664,8 @@ static void export_kernel_boot_props(void)
 
     for (i = 0; i < ARRAY_SIZE(prop_map); i++) {
         pval = property_get(prop_map[i].src_prop);
-        property_set(prop_map[i].dest_prop, pval ?: prop_map[i].def_val);
+        if (pval || prop_map[i].def_val)
+            property_set(prop_map[i].dest_prop, pval ?: prop_map[i].def_val);
     }
 
     pval = property_get("ro.boot.console");
-- 
1.7.4.1

